kernel/cost.cc

```diff
@@ -0,0 +1,226 @@
+#include "kernel/cost.h"
+
+USING_YOSYS_NAMESPACE
+
+unsigned int CellCosts::get(RTLIL::Module *mod)
+{
+	if (mod_cost_cache_.count(mod->name))
+		return mod_cost_cache_.at(mod->name);
+
+	unsigned int module_cost = 1;
+	for (auto c : mod->cells()) {
+		unsigned int new_cost = module_cost + get(c);
+		module_cost = new_cost >= module_cost ? new_cost : INT_MAX;
+	}
+
+	mod_cost_cache_[mod->name] = module_cost;
+	return module_cost;
+}
+
+static unsigned int y_coef(RTLIL::IdString type)
+{
+	// clang-format off
```

Comment:
Ew. This is convincing me more and more that we should not use clang-format...

Reply:
It's possible that it doesn't fit our repository. This really should have been a switch/case, but `ID($...)` isn't a constant value. Then I wouldn't have to do this. Personally I'm *very* used to hitting Ctrl+Shift+I to format an entire file I'm working on in VS Code. For shared files I intend to get used to formatting modified lines only, which VS Code does allow me to set a shortcut to

Reply:
You can use `type.in(...)` which may work better.